src/commands/protocols/finality/receiver/publish-finality-save-ack-command.js

```diff
@@ -20,7 +20,13 @@ class PublishFinalitySaveAckCommand extends Command {
      * @param command
      */
     async execute(command) {
-        const { ual, publishOperationId, blockchain, operationId, remotePeerId } = command.data;
+        const { ual, publishOperationId, blockchain, operationId, remotePeerId, state } =
+            command.data;
+
+        let ualWithState = ual;
+        if (state) {
+            ualWithState = `${ual}:${state}`;
+        }
```

Comment:
let { ual, publishOperationId, blockchain, operationId, remotePeerId, state } = command.data;

if (state) {
    ual = `${ual}:${state}`; // Modify ual directly
}

id do something like this, and also change command.data.ual to become the state with ual while doing continueSequence below, because if there's another command in the chain it would be good if it uses the state (or just in the command before this do not pass state but do if(state) ual+=':${state}'